### PR TITLE
Remove apparently dead code

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

**Comment by mwjames** [comment]
This is now a bit dangerous because this code is currently not tested and removing it before 2.1 release could make the release unstable without knowing.

File: includes/storage/SQLStore/SMW_SQLStore3_Queries.php [hunk 1]
```diff
@@ -858,53 +858,45 @@ protected function compileValueDescription(
 
 		// Try comparison based on value field and comparator,
 		// but only if no join with SMW IDs table is needed.
-		if ( $diType != SMWDataItem::TYPE_WIKIPAGE ) {
```
Comment (mwjames) [inline includes/storage/SQLStore/SMW_SQLStore3_Queries.php]: Where did this go?

https://scrutinizer-ci.com/g/SemanticMediaWiki/SemanticMediaWiki/code-structure/master/class/%2Bglobal%5CSMWSQLStore3QueryEngine

Comment (JeroenDeDauw) [inline includes/storage/SQLStore/SMW_SQLStore3_Queries.php]: If the else branch is dead, then the if part always got execute, so removing this check should not change anything.

**Comment by JeroenDeDauw** [comment]
> this code is currently not tested

Really? Don't we have integration tests that execute queries?

> before 2.1 release could make the release unstable without knowing

Well yeah. How about we branch from the commit before then?

**Comment by mwjames** [comment]
> Really? Don't we have integration tests that execute queries?

Yes, and we do have quite a few of them but `SQLStore3_Queries` is such mess that any changes can still have hidden effects that currently are not detected by our tests.

> Well yeah. How about we branch from the commit before then?

Probably the best idea.

**Comment by JeroenDeDauw** [comment]
You can haz `2.1.x` branch: https://github.com/SemanticMediaWiki/SemanticMediaWiki/branches

**Comment by mwjames** [comment]
Master is currently broken.

**Comment by JeroenDeDauw** [comment]
How do you figure that?

**Comment by mwjames** [comment]
> How do you figure that?

Was just looking at the red Travis button ;-)

**Pull request merged**